### PR TITLE
Issue #1362: Dockerize website generation and update readme

### PR DESCRIPTION
**Pull request by blakeflei**
# Description

Update to a sharable, working Docker implementation of reports that should run reliably without any changes to the underlying codebase. A single Docker image and docker-compose.yml configures a container that can be used to both generate reports and create the website. It will also serve the website locally from the container to verify completion. This PR addresses the [comment](https://github.com/cal-itp/data-infra/issues/1362#issuecomment-1117706754) from [#data-infra 1362](https://github.com/cal-itp/data-infra/issues/1362)

Some additional updates:
* A local install of the google cloud CLI was added to the Docker image now that we have non-destructive `gsutil rsync` commands in the readme. 
* Docker compose is configured so the website generated in the container is available via forwarded port 8000. 
* All files are either in the Docker image or stored locally, so the`-rm` flag was added to the docker-compose command so containers are removed when exited - nothing of any importance should remain in the container during normal use.

There are updates to the readme:
* For running the website generation in Docker
* For selectively rerunning the reports that don’t get generated each time. Incomplete report generation appeared to be more frequent when the number of threads selected is high (i.e. `make all -j30`), but it was also noticed this was a problem with fewer (i.e. `8`) threads. A note was added about using the `find` command to identify the folders that need manual re-running via papermill along with an example. This should address the issues mentioned [in the comments of a prior PR](https://github.com/cal-itp/reports/pull/114#issuecomment-1101515730), and make reports easier to maintain going forward. It could also be incorporated into a bash or python script going forward, but that was out of scope for this PR.
* Changed the order of the destructive `gsutil rsync` to gcloud buckets.

This PR resolves # [data-infra 1362](https://github.com/cal-itp/data-infra/issues/1362)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

## How has this been tested?
This was tested locally, but reports generation should be easily replicated locally via the following (this info is also in updated readme):
1. Install Docker locally, set resources as desired (i.e. enable 6 cores if you have an 8 core machine, etc.)
2. Clone repo and navigate to the repo root folder in a terminal, then follow the instructions in the readme.

### CODE REVIEW

**Comment by blakeflei** [comment]
Is our workflow merging to `main` or `development-build`? I can base it off either, but I noticed our [latest merge](https://github.com/cal-itp/reports/pull/119) was to main.

File: README.md [hunk 1]
```diff
@@ -93,50 +98,73 @@ Then, start the report generation:
 
 ```shell
 make generate_parameters
-make all -j 30
+make all -j 15
```
Comment (Nkdiaz) [inline README.md:101]: @blakeflei Why change the parallel processing to 15 when it can handle 30?

**Comment by acouch** [comment]
I was able to run this locally, outside the docker container by running:

`docker-compose run --rm --service-ports calitp_reports make -C reports MONTH=05 all`

I added some additional info to the README.md to better highlight the internals of the build process.

**Comment by edasmalchi** [comment]
thanks everyone for the helpful dialogue -- excited to merge this in!

**Pull request merged**